src/subclass/container.rs

```diff
@@ -51,65 +51,59 @@ impl<T: ContainerImpl + ObjectImpl> ContainerImplExt for T {
         unsafe {
             let data = self.get_type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkContainerClass;
-            let f = (*parent_class)
-                .add
-                .expect("No parent class impl for \"add\"");
-            f(container.to_glib_none().0, widget.to_glib_none().0)
+            if let Some(f) = (*parent_class).add {
+                f(container.to_glib_none().0, widget.to_glib_none().0)
+            }
         }
     }
 
     fn parent_remove(&self, container: &Container, widget: &Widget) {
         unsafe {
             let data = self.get_type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkContainerClass;
-            let f = (*parent_class)
-                .remove
-                .expect("No parent class impl for \"remove\"");
-            f(container.to_glib_none().0, widget.to_glib_none().0)
+            if let Some(f) = (*parent_class).remove {
+                f(container.to_glib_none().0, widget.to_glib_none().0)
+            }
         }
     }
 
     fn parent_check_resize(&self, container: &Container) {
         unsafe {
             let data = self.get_type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkContainerClass;
-            let f = (*parent_class)
-                .check_resize
-                .expect("No parent class impl for \"check_resize\"");
-            f(container.to_glib_none().0)
+            if let Some(f) = (*parent_class).check_resize {
+                f(container.to_glib_none().0)
+            }
         }
     }
 
     fn parent_set_focus_child(&self, container: &Container, widget: Option<&Widget>) {
         unsafe {
             let data = self.get_type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkContainerClass;
-            let f = (*parent_class)
-                .set_focus_child
-                .expect("No parent class impl for \"set_focus_child\"");
-            f(container.to_glib_none().0, widget.to_glib_none().0)
+            if let Some(f) = (*parent_class).set_focus_child {
+                f(container.to_glib_none().0, widget.to_glib_none().0)
+            }
         }
     }
 
     fn parent_child_type(&self, container: &Container) -> glib::Type {
         unsafe {
             let data = self.get_type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkContainerClass;
-            let f = (*parent_class)
-                .child_type
-                .expect("No parent class impl for \"child_type\"");
-            from_glib(f(container.to_glib_none().0))
+            if let Some(f) = (*parent_class).child_type {
+                from_glib(f(container.to_glib_none().0))
```

Comment:
This ain't compiling, go me. Also I don't think it makes sense for it to fallback to G_TYPE_NONE instead of panicking, this should really not be missing.

Reply:
The docs say:
> Note that this may return G_TYPE_NONE to indicate that no more children can be added

So yes, return `G_TYPE_NONE` here as a default :)

Reply:
See also https://gitlab.gnome.org/GNOME/gtk/blob/gtk-3-24/gtk/gtkcontainer.c#L1036-1051